vstools/exceptions/color.py

```diff
@@ -60,7 +60,8 @@ def check(func: FuncExceptT, to_check: vs.VideoNode) -> None:
         """
 
         try:
-            to_check.get_frame(0)
+            with to_check.get_frame(0):
+                pass
```

Comment:
Considering this is already inside a `try` block, it might be cleaner to use an explicit `finally` instead?
```py
        try:
            to_check.get_frame(0)
        except vs.Error as e:
        # -- snip --
        finally:
            to_check.close()
```

Reply:
This doesn't work as suggested. Works if I do something like this instead, though:

```py
        try:
            frame = to_check.get_frame(0)
        except vs.Error as e:
            if 'no path between colorspaces' in str(e):
                raise InvalidColorspacePathError(func, e)
            raise
        finally:
            frame.close()
```

Reply:
oh, whoops. yeah, that's true

Reply:
Yes, `close` needs to be on the returned frame, not the clip.

But: the adjusted snippet fails if `get_frame` actually throws, because then the assignment never happens.